MPXPLAY/AU_CARDS/SC_ICH.C

```diff
@@ -56,19 +63,18 @@
 
 #define ICH_PO_BDBAR_REG  0x10  // PCM out buffer descriptor BAR
 #define ICH_PO_LVI_REG    0x15  // PCM out Last Valid Index (set it)
-#define ICH_PO_CIV_REG    0x14  // PCM out current Index value (RO?)
-#define ICH_PO_PICB_REG   0x18  // PCM out position in current buffer(RO) (remaining, not processed pos)
+#define ICH_PO_CIV_REG    0x14  // PCM out current Index value (RO)
+#define ICH_PO_PICB_REG   ((card->device_type==DEVICE_SIS)?0x16:0x18)  // PCM out position in current buffer(RO) (remaining, not processed pos)
 
 #define ICH_ACC_SEMA_REG  0x34  // codec write semiphore register
 #define ICH_CODEC_BUSY    0x01  // codec register I/O is happening self clearing
 
 #define ICH_BD_IOC        0x8000 //buffer descriptor high word: interrupt on completion (IOC)
 
-#define ICH_DMABUF_MAX_PERIODS  32
-#define ICH_DMABUF_PERIODS   4
-#define ICH_MAX_CHANNELS     2
-#define ICH_MAX_BYTES        4
-#define ICH_DMABUF_ALIGN (ICH_DMABUF_MAX_PERIODS*ICH_MAX_CHANNELS*ICH_MAX_BYTES) // 256
+#define ICH_DMABUF_MAX_PERIODS  32 // number of entries in the Buffer Descriptor List
+#define ICH_DMABUF_PERIODS   4 // number of "used" entries in the Buffer Descriptor List
+#define ICH_BDL_ENTRY_SIZE (2 * sizeof(uint32_t))  // size of one entry in the Buffer Descriptor List
+#define ICH_DMABUF_ALIGN (ICH_DMABUF_MAX_PERIODS*ICH_BDL_ENTRY_SIZE) // 256
```

Comment:
This is kind of unrelated to SIS7012 support, but during reading of the AC97 docs on OSDev, I found that some of these things look mislabeled (it works because 2 * sizeof(uint32_t) is the same as 2 * 4, even though the "2" doesn't have anything to do with channels. Those values are used for the Buffer Descriptor List.